tests/api_workflow/test_api_workflow_datasources.py

```diff
@@ -142,19 +147,95 @@ def test_download_raw_sample_metadata(self):
         assert len(predictions) == num_samples
         pbar.update.assert_called()
 
     def test_download_raw_samples_predictions_relevant_filenames(self):
         self.api_workflow_client._datasources_api.reset()
-        predictions = self.api_workflow_client.download_raw_predictions('test', relevant_filenames_file_name="test")
+        predictions = self.api_workflow_client.download_raw_predictions(
+            "test", relevant_filenames_file_name="test"
+        )
         num_samples = self.api_workflow_client._datasources_api._num_samples
         assert len(predictions) == num_samples
 
     def test_get_prediction_read_url(self):
         self.api_workflow_client._datasources_api.reset()
-        read_url = self.api_workflow_client.get_prediction_read_url('test.json')
+        read_url = self.api_workflow_client.get_prediction_read_url("test.json")
         self.assertIsNotNone(read_url)
+
+    def test_duplicate_filenames(self):
+        self.api_workflow_client._datasources_api.reset()
+        self.api_workflow_client._datasources_api._samples = defaultdict(
+            lambda: [
+                DatasourceRawSamplesDataRow(file_name="file_0", read_url="url_0"),
+                DatasourceRawSamplesDataRow(file_name="file_0", read_url="url_0"),
+                DatasourceRawSamplesDataRow(file_name="file_1", read_url="url_1"),
+                DatasourceRawSamplesDataRow(file_name="file_2", read_url="url_2"),
+                DatasourceRawSamplesDataRow(file_name="file_3", read_url="url_3"),
+                DatasourceRawSamplesDataRow(file_name="file_4", read_url="url_4"),
+            ]
+        )
+        with self.assertWarns(UserWarning) as context:
+            samples = self.api_workflow_client.download_raw_samples()
+        self.assertTrue(
+            "Duplicate filename file_0 in relevant filenames file"
+            in str(context.warning)
+        )
+
+        assert len(samples) == 5
+        # expect a warning
+
+    def test_absolute_filenames(self):
+        self.api_workflow_client._datasources_api.reset
+        self.api_workflow_client._datasources_api._samples = defaultdict(
+            lambda: [
+                DatasourceRawSamplesDataRow(file_name="/file_0", read_url="url_0"),
+                DatasourceRawSamplesDataRow(file_name="file_1", read_url="url_1"),
+                DatasourceRawSamplesDataRow(file_name="file_2", read_url="url_2"),
+                DatasourceRawSamplesDataRow(file_name="file_3", read_url="url_3"),
+                DatasourceRawSamplesDataRow(file_name="file_4", read_url="url_4"),
+            ]
+        )
+        with self.assertRaises(ValueError) as context:
+            samples = self.api_workflow_client.download_raw_samples()
+        self.assertTrue(
+            "Absolute file paths like /file_0 are not supported in relevant filenames file"
+            in str(context.exception)
+        )
+
+    def test_dot_slash(self):
```

Comment:
We decide to use the naming "test_{name_of_function_to_be_tested}__special_case_to_be_tested"

Reply:
Also for the other tests added.

Reply:
- Doc updated to explicitly instruct against using dot notation (see screenshot too). 
- Adopted new pytest syntax for asserting warnings/ errors.
- Changed test names to match tested method names.